app/controllers/users/edit_info_controller.rb

```diff
@@ -46,13 +46,20 @@ def process_updates(resource)
       updater = UserFlashUpdater.new(@update_form, flash)
       updater.set_flash_message
 
-      if @update_form.phone_changed?
-        analytics.track_event('User asked to update their phone number')
-        prompt_to_confirm_phone(@update_form.phone)
+      track_analytics
+
+      if @update_form.require_phone_confirmation?
+        prompt_to_confirm_phone(@update_form.phone,
+                                @update_form.phone_sms_enabled)
       elsif is_flashing_format?
         EmailNotifier.new(resource).send_password_changed_email
         redirect_to profile_url
       end
     end
+
+    def track_analytics
+      analytics.track_event('User asked to update their phone number') if
+        @update_form.phone_changed?
+    end
   end
 end
```

Comment:
This controller was removed last week. It looks like you added it back when you rebased #375. I just opened a new PR to remove it. Please merge #386, then rebase against master to remove this controller from this PR. Thanks!